clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2087,6 +2087,7 @@ class SyclKernelPointerHandler : public SyclKernelFieldHandler {
 
     const ConstantArrayType *CAT =
         SemaRef.getASTContext().getAsConstantArrayType(ArrayTy);
+    assert(CAT && "Should only be called on constant-size array.");
```

Comment:
Same comment as in the earlier PR for enterArray. This cannot be null because we do a check when we enter these handlers for processing complex arrays. Since the PR for enterArray was merged in I guess for consistency this can be merged in as well. But IMO both this and the earlier patch was not required and  static analyzer issue could have been fixed by marking it as not an issue.

Reply:
Thank you @elizabethandrews for the comments.